### PR TITLE
Get index roots from Wikidata

### PR DESCRIPTION
**Pull request by ricordisamoa**
Not tested.

### CODE REVIEW

File: build.php [hunk 1]
```diff
@@ -19,13 +19,20 @@
...
 foreach ($dbs as $lang => $info) {
-    $dsn = $info['dsn'];
-    $indexNs = $info['index_ns'];
-    $indexRoot = $info['index_root'];
-    $catLabel = $info['cat_label'];
-    $pdo = new PDO($dsn, $dbuser, $dbpass);
-    buildOneLang( $pdo, $lang, $indexRoot, $catLabel, $indexNs );
+    $dbName = $info['db_name'];
+    if (isset($indexRoots[$dbName])) {
+        $dsn = "mysql:dbname={$dbName}_p;host=$dbName.labsdb";
```
Comment (samwilson) [inline build.php:30]: I didn't want to construct the DSN like this, only because it makes it harder for local testing whilst in development.

**Comment by samwilson** [comment]
Thanks for this, it's a great idea! I'm nearly knocking off for the evening now; will delve deeper tomorrow. :)

**Comment by samwilson** [comment]
Okay, have a look now. I've rejigged things a bit, so it builds all required site info _unless_ it's provided in `config.php`. I think this works.

Thank you so much for showing me how to make this all be possible! I'd not really investigated multi-language page links in Wikidata before.

It's also looking up the `Index` namespace IDs from the individual Wikisources.

I'm not quite sure that all is as it should be, because `cs` is returning _nothing_...

**Comment by ricordisamoa** [comment]
Thanks!

**Pull request merged**